lib/galaxy/tools/evaluation.py

```diff
@@ -513,13 +513,13 @@ def _build_config_files(self):
             config_text, is_template = self.__build_config_file_text(content)
             # If a particular filename was forced by the config use it
             directory = ensure_configs_directory(self.local_working_directory)
-            with tempfile.NamedTemporaryFile(dir=directory, delete=False) as temp:
-                config_filename = temp.name
             if filename is not None:
-                # Explicit filename was requested, this is implemented as symbolic link
-                # to the actual config file that is placed in tool working directory
+                # Explicit filename was requested, needs to be placed in tool working directory
                 directory = os.path.join(self.local_working_directory, "working")
-                os.link(config_filename, os.path.join(directory, filename))
+                config_filename = os.path.join(directory, filename)
```

Comment:
This logic is no longer the same. The `os.link` call to the temp file is missing? Why was this change necessary?